frontend/src/presentation/components/QuestionFilters.tsx

```diff
@@ -36,93 +36,90 @@ export const QuestionFilters: React.FC<QuestionFiltersProps> = ({
+            searchTerm,
+        });
     };
 
-    const handleCategoryTagChange = useCallback(
-        (category: string, checked: boolean) => {
-            const nextSelectedCategories = checked
-                ? [...selectedCategories, category]
-                : selectedCategories?.filter((c) => c !== category);
-            setSelectedCategories(nextSelectedCategories);
-            triggerFiltersChange({
-                selectedDifficulty,
-                selectedCategories: nextSelectedCategories,
-                searchTerm,
-            });
-        },
-        [selectedDifficulty, selectedCategories, searchTerm, triggerFiltersChange]
-    );
+    const handleAddCategory = async (category: string) => {
+        if (!category || category.trim() === "") {
+            message.error("Category cannot be empty!");
+            return;
+        }
 
-    const filteredCategories = useMemo(() => {
-        const lowerSearchTerm = categorySearchTerm.toLowerCase();
-        return allCategories.filter((category) =>
-            category.toLowerCase().includes(lowerSearchTerm)
-        );
-    }, [allCategories, categorySearchTerm]);
+        if (allCategories.includes(category)) {
+            message.error("Category already exists!");
+            return;
+        }
 
-    const categoryDropdownContent = (
-        <div
-            className={styles.categoryDropdownContent}
-            onClick={(e) => e.stopPropagation()}
-        >
-            <div className={styles.categoryDropdownContainer}>
-                <Input
-                    placeholder={QUESTIONS_FILTER_TEXT.SELECT_CATEGORIES}
-                    value={categorySearchTerm}
-                    onChange={handleCategorySearch}
-                    className={styles.categorySearchBar}
-                    allowClear
-                />
-                <div className={styles.categoriesGrid}>
-                    {filteredCategories.map((category) => (
-                        <CheckableTag
-                            key={category}
-                            checked={selectedCategories.includes(category)}
-                            onChange={(checked) => handleCategoryTagChange(category, checked)}
-                        >
-                            {category}
-                        </CheckableTag>
-                    ))}
-                </div>
-            </div>
-        </div>
+        try {
+            await categoryUseCases.createCategory(category);
+            setAllCategories([...allCategories, category]);
+            message.success("Category added successfully!");
+        } catch (error) {
+            message.error("Failed to add category!");
+            console.error("Failed to add category:", error);
+        }
+    };
+
+    const handleDeleteCategory = async (categoriesToDelete: string[]) => {
+        try {
+            await Promise.all(categoriesToDelete.map((category) => categoryUseCases.deleteCategory(category)));
```

Comment:
I think we need BE to have bulk deletion too haha. Can deprioritise though

Reply:
no need lah, a bit too troublesome, and it isn't expected for the deletion to have too many entries